protostar/testing/cheatcodes/deploy_cheatcode.py

```diff
@@ -56,8 +60,27 @@ def deploy_prepared(
         return DeployedContract(contract_address=prepared.contract_address)
 
     def invoke_constructor(self, prepared: PreparedContract):
+
+        self.validate_constructor_args(prepared)
         self.execute_constructor_entry_point(
             class_hash_bytes=to_bytes(prepared.class_hash),
             constructor_calldata=prepared.constructor_calldata,
             contract_address=prepared.contract_address,
         )
+
+    def validate_constructor_args(self, prepared: PreparedContract):
+        contract_class = self.state.get_contract_class(to_bytes(prepared.class_hash))
+
+        if not contract_class.abi:
+            raise CheatcodeException(self, "Contract ABI was not found")
```

Comment:
```suggestion
            raise CheatcodeException(self, "Contract ABI was not found. Unable to verify constructor arguments.")
```

+ maybe include information which contract is problematic?